### PR TITLE
Explore relationship model is capturing in encoder

### PR DESCRIPTION
**Pull request by ajlee21**
The main change for this PR is the creation of the explore_relationship_AandB_pt2.ipynb notebook that is a followup of the previous explore_relationship_AandB_pt1.ipynb nb (renamed from explore_relationship_AandB.ipynb).

In this notebook we artificially shift the input data and look to see what transformed/predicted gene expression data looks like in order to determine if our nonlinear signal that was trained on is learned.

### CODE REVIEW

File: scripts/nbconverted/explore_relationship_AandB_pt2.py [hunk 1]
```diff
@@ -0,0 +1,682 @@
...
+
+sim_data_file = os.path.join(base_dir, "data", analysis_name, "train_model_input.txt.xz")
+A_file = os.path.join(base_dir, "data", analysis_name, "geneSetA.txt")
+B_file = os.path.join(base_dir, "data", analysis_name, "geneSetB.txt")
+
+offset_vae_file = os.path.join(os.path.dirname(os.getcwd()), "encoded", analysis_name, "offset_latent_space_vae.txt")
+
+weight_file = os.path.join(os.path.dirname(os.getcwd()), "data", analysis_name, "VAE_weight_matrix.txt")
+
+model_encoder_file = glob.glob(os.path.join(base_dir, "models", analysis_name, "*_encoder_model.h5"))[0]
```
Comment (danich1) [inline scripts/nbconverted/explore_relationship_AandB_pt2.py]: Would prefer these long lines to be broken up into a version with more white space:
```python
glob.glob(
  os.path.join(
  base_dir, "models", 
  analysis_name, "*_encoder_model.h5"
  )
)[0]
```

File: scripts/nbconverted/explore_relationship_AandB_pt2.py [hunk 2]
```diff
@@ -0,0 +1,682 @@
...
+    
+    sample_ids = sample_data.index
+    for sample_id in sample_ids:
+        sample_exp = sample_data.loc[sample_id].to_frame().T
+        
+        # Use trained model to encode expression data into SAME latent space
+        predict = loaded_model.predict_on_batch(sample_exp)
+
+        predict_encoded_df = pd.DataFrame(predict)
+        predicted_encoded_sample_data = predicted_encoded_sample_data.append(predict_encoded_df, ignore_index=True)
```
Comment (danich1) [inline scripts/nbconverted/explore_relationship_AandB_pt2.py]: Up to you, but I highly recommend changing lines like this into:
```python
predicted_encoded_sample_data = (
  predicted_encoded_sample_data
  .append(predict_encoded_df, ignore_index=True)
)
```
This works for any line that has multiple or single functions being called.

**Review by danich1 — APPROVED** [review]
Great PR @ajlee21. Majority of the review has been talked about in person. Three points of recap:
1. There was an issue about how the scale factor (alpha) is being calculated. Original design is having the scale factor be calculated in the gene expression space. This might not translate well into transforming latent space features and the suggestion here is to find a way to calculate alpha in the latent space.
2. Instead of using all the features for calculating the offset vector, just use only the highly informative features. 
3. Highly recommend doing a simple proof of concept experiment where the data matrix is pruned to only include group A and B genes. Using this pruned matrix, do the latent space transformation and see if the simulated signal is captured. If success, then consider throwing in a small group of unaffected genes and see if the signal remains. Continue until the signal gets drowned out. (Basically an ablation study for your simulated signal.)

I approve of this PR given that you make the code design changes mentioned above.

**Comment by ajlee21** [comment]
Thank you @danich1 Made formatting changes.  Will address the additional comments in another PR.

**Pull request merged**